cmd/check_container.go

```diff
@@ -186,15 +186,51 @@ var checkContainerCmd = &cobra.Command{
 				)
 			}
 
+			certImage, err := os.Open(path.Join(artifacts.Path(), certification.DefaultCertImageFilename))
+			if err != nil {
+				return fmt.Errorf(
+					"%w: could not open file for submission: %s",
+					err,
+					certification.DefaultCertImageFilename,
+				)
+			}
+			preflightResults, err := os.Open(path.Join(artifacts.Path(), certification.DefaultTestResultsFilename))
+			if err != nil {
+				return fmt.Errorf(
+					"%w: could not open file for submission: %s",
+					err,
+					certification.DefaultTestResultsFilename,
+				)
+			}
+			rpmManifest, err := os.Open(path.Join(artifacts.Path(), certification.DefaultRPMManifestFilename))
+			if err != nil {
+				return fmt.Errorf(
+					"%w: could not open file for submission: %s",
+					err,
+					certification.DefaultRPMManifestFilename,
+				)
+			}
+			logfile, err := os.Open(viper.GetString("logfile"))
+			if err != nil {
+				return fmt.Errorf(
+					"%w: could not open file for submission: %s",
+					err,
+					viper.GetString("logfile"),
+				)
+			}
```

Comment:
In fairness, passing the file-reading logic to the builder meant this check_container was easier to read because it didn't have these 4 opens, and associated error handling. Ah well. the testing is better. I'll take that.

That said, are we missing the `defer (*File).close()` calls here?

Reply:
As I read this, I realized that was missing.